src/main/java/ch/cern/eam/wshub/core/services/entities/UserDefinedFields.java

```diff
@@ -1735,96 +1732,107 @@ public String toString() {
-				&& udfObjectComparison.test(getUdfchar42(), other.getUdfchar42())
-				&& udfObjectComparison.test(getUdfchar43(), other.getUdfchar43())
-				&& udfObjectComparison.test(getUdfchar44(), other.getUdfchar44())
-				&& udfObjectComparison.test(getUdfchar45(), other.getUdfchar45())
-				&& udfObjectComparison.test(getUdfchkbox01(), other.getUdfchkbox01())
-				&& udfObjectComparison.test(getUdfchkbox02(), other.getUdfchkbox02())
-				&& udfObjectComparison.test(getUdfchkbox03(), other.getUdfchkbox03())
-				&& udfObjectComparison.test(getUdfchkbox04(), other.getUdfchkbox04())
-				&& udfObjectComparison.test(getUdfchkbox05(), other.getUdfchkbox05())
-				&& udfObjectComparison.test(getUdfchkbox06(), other.getUdfchkbox06())
-				&& udfObjectComparison.test(getUdfchkbox07(), other.getUdfchkbox07())
-				&& udfObjectComparison.test(getUdfchkbox08(), other.getUdfchkbox08())
-				&& udfObjectComparison.test(getUdfchkbox09(), other.getUdfchkbox09())
-				&& udfObjectComparison.test(getUdfchkbox10(), other.getUdfchkbox10())
-				&& udfObjectComparison.test(getUdfnum01(), other.getUdfnum01())
-				&& udfObjectComparison.test(getUdfnum02(), other.getUdfnum02())
-				&& udfObjectComparison.test(getUdfnum03(), other.getUdfnum03())
-				&& udfObjectComparison.test(getUdfnum04(), other.getUdfnum04())
-				&& udfObjectComparison.test(getUdfnum05(), other.getUdfnum05())
-				&& udfObjectComparison.test(getUdfnum06(), other.getUdfnum06())
-				&& udfObjectComparison.test(getUdfnum07(), other.getUdfnum07())
-				&& udfObjectComparison.test(getUdfnum08(), other.getUdfnum08())
-				&& udfObjectComparison.test(getUdfnum09(), other.getUdfnum09())
-				&& udfObjectComparison.test(getUdfnum10(), other.getUdfnum10())
-				&& udfObjectComparison.test(getUdfdate01(), other.getUdfdate01())
-				&& udfObjectComparison.test(getUdfdate02(), other.getUdfdate02())
-				&& udfObjectComparison.test(getUdfdate03(), other.getUdfdate03())
-				&& udfObjectComparison.test(getUdfdate04(), other.getUdfdate04())
-				&& udfObjectComparison.test(getUdfdate05(), other.getUdfdate05())
-				&& udfObjectComparison.test(getUdfdate06(), other.getUdfdate06())
-				&& udfObjectComparison.test(getUdfdate07(), other.getUdfdate07())
-				&& udfObjectComparison.test(getUdfdate08(), other.getUdfdate08())
-				&& udfObjectComparison.test(getUdfdate09(), other.getUdfdate09())
-				&& udfObjectComparison.test(getUdfdate10(), other.getUdfdate10());
+		return getters.stream().anyMatch(func -> {
+			Object changerValue = func.apply(modification);
+			return changerValue != null && (original == null || changerValue != func.apply(original));
+		});
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		UserDefinedFields that = (UserDefinedFields) o;
+		return getters.stream().allMatch(getter -> Objects.equals(getter.apply(this), getter.apply(that)));
```

Comment:
Two safeties with the same UDFs but different hazards/precautions are equivalent?